syntax = "proto3";
package summa.proto;

// Analyzes indices
service ReflectionApi {
  rpc get_top_terms (GetTopTermsRequest) returns (GetTopTermsResponse) {}
}

// Requests

message GetTopTermsRequest {
  string index_name = 1;
  string field_name = 2;
  uint32 top_k = 3;
}

// Extra structures

message GetTopTermsResponse {
  map<string, SegmentTerms> per_segment = 1;
}

message SegmentTerms {
  repeated TermInfo term_infos = 1;
}

message TermInfo {
  bytes key = 1;
  uint32 doc_freq = 2;
}
